api/v1/policy_webhook.go

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package v1
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// log is for logging in this package.
+var policylog = logf.Log.WithName("policy-resource")
+
+func (r *Policy) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-policy-open-cluster-management-io-v1-policy,mutating=false,failurePolicy=fail,sideEffects=None,groups=policy.open-cluster-management.io,resources=policies,verbs=create,versions=v1,name=policy.open-cluster-management.io.webhook,admissionReviewVersions=v1
+
+var _ webhook.Validator = &Policy{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *Policy) ValidateCreate() (admission.Warnings, error) {
+	policylog.Info("validate create", "name", r.Name)
+
+	return r.validate()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *Policy) ValidateUpdate(_ runtime.Object) (admission.Warnings, error) {
+	policylog.Info("validate update", "name", r.Name)
+
+	return nil, nil
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *Policy) ValidateDelete() (admission.Warnings, error) {
+	policylog.Info("validate delete", "name", r.Name)
+
+	return nil, nil
+}
+
+// validate admits a pod if a specific annotation exists.
+func (r *Policy) validate() (admission.Warnings, error) {
+	policylog.Info("Validating Policy by webhook")
+
+	namespaceLen := len([]rune(r.GetNamespace()))
```

Comment:
@yiraeChristineKim you need to ensure this only runs on root policies, otherwise, this check will also erroneously apply to replicated policies.

For example, if my root policy is in the namespace `open-cluster-management-global-set` and the policy name is `policy-name`, the webhook would validate the length of `open-cluster-management-global-set.policy-name`.

Then say the policy applies to a cluster named `very-long-cluster-name-for-this-example`, then the webhook would erroneously check that the length of `very-long-cluster-name-for-this-example.open-cluster-management-global-set.policy-name`.

Reply:
Now I fully understand!